### PR TITLE
Anon RingCT Transactions Only

### PR DESCRIPTION
**Pull request by particlmike33**
This Proposal is for making the Particl network anon (ringct) transactions only.  No longer would there be an option for public transactions or blind transactions (ct).

With opt-in privacy (like it is now), the devs' work is undermined by the fact that when most people don't opt-in, ringct doesn't do much for the people that do use it.  The anonymity set tanks with opt-in privacy.

If we want the full benefit from the devs' hard work, we need to have only anon transactions.

Our "thing" as a project is a whole suite of privacy tools.  We are a privacy coin.  So we should be as anonymous as possible without the user needing to do anything.  And that's how we'll attract more people to Particl.

We are advertising Particl based on it's superior privacy qualities.  When people come to find out Particl isn't as private as it could be, they will be hesitant to use it.  And we want as many people to join Particl as possible.  That will happen when Particl has the best privacy possible, with only anon transactions.

Unfortunately, Particl over the last few years has mostly slid out of people's view even with all the groundbreaking tech our devs code.  Our devs make more updates than most other projects do, yet most people sadly still don't know we exist.

There's a huge disconnect between how great our code and devs are, and how little people know about us.

Making all transactions anon (ringct) would boost our privacy and anonymity massively, help us get back to our roots of being a go-to privacy coin, and help us get back on the map for the wider crypto community.

Monero did this awhile back and it really helped them.  And they don't even have a marketplace or dex!  And they are based on cryptonote instead of bitcoin like we are!  Yet their marketcap is 200x larger than ours, and daily volume traded is 38,000x larger.

Do you really think we should be valued so much less than monero?  I think not, and having privacy on par/better than monero will help us be valued much greater.

The people that will flock to our community care about privacy, and having opt-in privacy is a huge flaw that hurts us.
We should be the best we can be, and with only anon transactions, and a marketplace and dex, we would be unstoppable.

The monero subreddit was recently discussing our dex, and while they seemed mostly excited, a couple people did bring up how we only have opt-in privacy, and therefore are not a good privacy project (said in much harsher words).

This shows that people considering joining particl want privacy, and the best privacy possible.  Privacy concerned people don't want to join a system that has flaws that could be simply fixed, like opt-in privacy transactions instead of only anon transactions.  If we are as private as possible, then we have a better chance of people becoming interested in Particl.  Without that, they will turn away just like those people in the monero subreddit did upon finding out our privacy is opt-in.

For anyone that wants opt-in privacy: Why do you want the project to not be as anonymous and secure as possible?  You can't have it both ways, having opt-in privacy hurts the private transactions.  They are not independent of each other.  They are not separate blockchains.

Why just go half way with our privacy tech when the tech is there to go all the way?  With opt-in privacy, we are just getting a bit of the benefits we could be getting from the inventions the dev team have created.

Should we also campaign Tor to be faster but less anonymous, or monero to be more scalable, but less anonymous?  Or Signal Messenger to be better usable but less securely encrypted?

It just doesn't make sense, secure systems are built with security in mind, usability second.

With opt-in privacy we are making security second and usability first instead.
We are going away from our original ethos, a secure and private system.
That is what Particl has always been about.
No one is going to want to buy a privacy coin that isn't that private compared to its competitors.
No one is going to want to buy a privacy coin that has skimped on it's privacy settings and abilities.

To be competitive, and to be as secure and private as possible, we need to have only anon transactions.

This proposal is for a vote.  If most of the community agrees, then we will pass it on to the devs.  I am a community member, not a dev.

### CODE REVIEW

**Comment by particlmike33** [comment]
Making this update wouldn't require massive changes to how Particl works.

Staking would still be public, but the first transaction afterwards would be required to be a convert to anon transaction (like how zcash does).

Anon transactions don't work for escrow, so the marketplace currently handles that by requiring a ringct (anon) transaction to a ct (blind) escrow address, and then transactions from that escrow address are also required to be ringct (anon).

This requirement could be made the same for non marketplace transactions.  The only time blind transactions are used are for escrow addresses, and they require an anon transaction on each side of it.

All other transactions are anon only, just like the marketplace.

Basically, by adopting the rules we have for the marketplace and setting them for the entire Particl network, the anonymity and privacy of our coin would be greatly increased

**Comment by coybigpart** [comment]
How much time and funds do you need to realize this?

**Comment by tecnovert** [comment]
This makes no sense to me.
The system works as is.

**Comment by particlmike33** [comment]
> This makes no sense to me. The system works as is.

It does work now but the network would be more anonymous with these changes.  Why would we not want better/more secure anonymity?

Don't you agree that opt-in anonymity isn't as secure compared to all the transactions being ringct?

It's the reason chainalysis can track zcash and dash but not monero.

**Comment by particlmike33** [comment]
> How much time and funds do you need to realize this?

I don't know, the devs would need to comment on that.

I was more making this proposal to gauge community support for heading in this direction with only anon transactions, (essentially adopting the rules for the marketplace to the entire network).

If a lot of community members like the idea, it may be added to the roadmap.

**Comment by coybigpart** [comment]
@particlmike33 feel free to fork Particl and make it anon only.

**Comment by particlmike33** [comment]
@coybigpart I don't want to, I want to see the community's thoughts on anon only transactions, which is the whole purpose of github pulls/issues: for people to comment on proposals and come up with ideas/directions a project wants to head in.

Some people commented positively on discord about increasing the anonymity set, and others didn't, so I created a community proposal, which is the entire point of our governance feature.

**Comment by coybigpart** [comment]
These kind of decisions shouldn't be made by the community, but by the Particl team.

**Comment by particlmike33** [comment]
@coybigpart Ultimately the final decision is up to the devs, but I disagree that our governance mechanism isn't meant for these types of things.  There's nothing wrong with gauging community interest in certain issues, and it's why we have github pulls and our governance model.

If you don't like my proposal that's fine and I'd welcome your opinions on why you want a smaller anonymity set, but disagreeing with the very notion of our community governance model doesn't make much sense to me, when it was already approved about a year ago.

**Comment by particlmike33** [comment]
@tecnovert @coybigpart Do you disagree that having opt-in privacy lowers our anonymity set?

Quotes from these resources below, show that having opt-in privacy is detrimental to the people that choose to use anon transactions, and the solution to that is to mandate ringct, and then transactions will actually be anonymous.

https://medium.com/digitalassetresearch/monero-becomes-bulletproof-f98c6408babf

"In Monero’s early history, users could potentially have 0 mixins in their transactions, or in other words, create a ring signature comprised solely of the true sender’s signature. Mining pools used 0 mixins when disbursing funds to constituents, which does not require privacy. However, as the Monero Research Labs originally found out, doing so hurts everyone else’s privacy."

^Monero used to have opt-in privacy like Particl, but researchers discovered this hurt the people that chose to use privacy.  They aren't unrelated and it isn't about removing an option for people to use public, it's about ensuring that people that want privacy actually have it.  And they don't with opt-in privacy because the public transactions hurt the attempted private ones.

https://ieeexplore.ieee.org/document/8456034

"Recent studies show that the majority of Monero inputs are traceable
prior to mandatory RingCT transaction. After the RingCT was implemented,
the problems were mitigated."

This is even more damning, and it proves that ringct transactions can be traced when there aren't enough of them.  This is the current situation with particl and our opt-in privacy.  Monero used to have the exact same issue and by mandating ringct, they solved the problem.

Why would we not want to do this too and further secure our network?

**Comment by bacoinin** [comment]
Its pointless to attempt to reason with trolls, they will ignore every piece of valid reasoning or information that is presented to them. This particular one seems to be failing at basic comprehension while presenting as "proof" to his idiotic claims text that he either does not understand or simply does not need to. I am strongly convinced of the latter and thus deduced that the OP is a simply a troll and not a very smart one. 

Take for example his first quote (taken from an article that comments on the privacy concerns in a single paragraph):

>"In Monero’s early history, users could potentially have 0 mixins in their transactions, or in other words, create a ring signature comprised solely of the true sender’s signature. Mining pools used 0 mixins when disbursing funds to constituents, which does not require privacy. However, as the Monero Research Labs originally found out, doing so hurts everyone else’s privacy."

The quote above was spamed on the general channel on Discord as  undeniable proof that "opt-in privacy doesn't work and is traceable" or "right now we are only anonymous to the level of your friend trying to trace your transaction, but for a company or any big researcher, we are not anonymous"... 

To my kind request explain on how he things this is related to Particl and the existance of public transactions (with a note those 0-mixins transactions on Monero not the same as public transactions on Particl) the troll replies:

"0 mixin transactions are more anonymous than our public transactions because they were blind transactions, if the flaw exists for blind transactions then it certainly exists for public transactions"

For the record lets see if what the troll claims in valid before examining if its sound.
Fact 1. Monero introduces minimum ring size = 3 in March 2016 and the blind transaction or more precisely RingCT is introduced in January 2017. 

So the claim that the 0-mixins were CT transactions is bogus...

Lets look at the next part of his claim
>"if the flaw exists for blind transactions then it certainly exists for public transactions" 
which is another pile of crap. 

If the OP had bothered to read the IEEE paper that he cites as the 2nd undeniable proof, he would have seen the one of the two papers on which it relies to draw its conclusion that:
>"Recent studies show that the majority of Monero inputs are traceable
prior to mandatory RingCT transaction." 

states the following on how the 0-mixin transactions reduce the privacy for all:

"Not only do 0-mixin transactions provide no privacy
to the users that created them, but also, more worryingly,
they present a privacy hazard if other users include these
provably-spent transaction outputs as mixins in other trans-
actions too. When the Monero client chooses mixins, it
does not take into account whether the potential mixins
have already been spent."

and that has nothing to do nor has any similarities with how the Particl's executed public transactions affect the Anon set...to be precise they practically don't. So 

Note: One could far stretch it and assume that user A does Public2Anon2Public and withdraws the entire amount and another user does the same with a ring size 2 and accidentally includes the visible spent output by the user A. This is a completely unrealistic scenario as it assumes that such custom transactions are done by clueless people that are actually interested in their privace... 

And before I stop feeding the troll, a small comment on the proposal itself. The OP fails to see how Particl aims to serves a very different purpose that Monery which aims to be digital cash. Particl aims to facilitate the development of privacy-centric dApps. Its blockchain is one of the back-end components that are used for that purpose (and others). The choice of an up-to-date Bitcoin Core codebase for its DLT component is near optimal one for man reasons e.g. programmability via Script, LNP, Taproot and many amazing features, features that Monero does not even dare to dream about.... 

As for the Anon set size, number of Anon transactions, etc aka obfuscation noise...Here is my prediction: 

at the most by the end of 2022 we will have a much better obfuscation/privacy than Monero due to the way that PM is designed and the atomic swaps DEX that will allow the users to enter/user the PM by directly swapping traceable coins like BTC/LTC directly to Anon PART bypassing the Public transactions that the CEXes currently use.

If Taproot is used on the BTC side too, then viola any transparent coin has top-notch privacy protection when shopping on the PM. 

Kind regards,
Mike

P.S. I wrote the above just for the public scrutiny and record and not to disagree with you, The answer to your proposal/suggestion is "No thank you"

“Never argue with an idiot. They will drag you down to their level and beat you with experience.”

― Mark Twain

**Comment by particlmike33** [comment]
Ridiculous how you accuse me of being a troll because I'm citing peer reviewed security papers....

What you wrote is just wrong.  A 0 mixing ringct transaction is a blind transaction!  If the flaw exists for that it certainly does for public transactions.

Let's say I'm wrong about 0 mixin ringct transactions being equivalent to blind transactions, and instead they are like public transactions.  Then I'm still right!  Because the authors note that public transactions hurt the anonymity of the ringct transactions, they do not exist in a vacuum.

You are just crossing your fingers and hoping that somehow having a MP means we will be exempt from this well documented flaw.

Please answer me this:

Are we going to monitor the percent of all transactions that are ringct every hour, and if it drops too low then notify everyone that their ringct transactions won't be anonymous anymore?
And if it is a high enough percentage, tell them that the public transactions are still hurting their privacy, but not too badly, and they can go ahead and send a ringct transaction if they want?
that seems crazy to me when instead we could just mandate ringct and fix the problem, which is exactly what professional cryptographers recommend doing in those 2 papers!

You like the fact that the MP mandates ringct, because it helps our privacy, but you don't like making those same rules for off MP transactions?  And you can't/refuse to explain why?

The flaw exists.  It's a fact.  It has affected every chain with opt-in privacy.  It is well documented by security researchers.

Just crossing your fingers and covering your ears and saying: well hopefully the MP will be 99%+ of all transactions so the flaw won't affect us, is just not logical!

Instead we could make the same rules we have for the MP for off MP transactions too, and then we can be sure that our ringct transactions will always be private, instead of hoping and needing to monitor the percent of all transactions that are ringct and hoping it always stays high enough that the flaw won't affect us too badly.

Because the truth is ALL public transactions hurt the privacy of ringct transactions.  If 99% of all transactions are ringct, then it might be negligible how much the 1% hurt our privacy.

But are you really going to monitor that every hour and cross your fingers that our ringct transactions stay private?  Instead of just mandating ringct across off MP transactions?

**Comment by coybigpart** [comment]
@particlmike33  "Do you disagree that having opt-in privacy lowers our anonymity set"

No I don't disagree with that. But I think once the dex and marketplace are fully used we have enough anon transactions to be very private. It's up to the users to choose the amount of decoys/privacy. By only allowing anon transactions we can not use the governance system anymore for example.

**Comment by coybigpart** [comment]
@particlmike33  "Because the truth is ALL public transactions hurt the privacy of ringct transactions. If 99% of all transactions are ringct, then it might be negligible how much the 1% hurt our privacy."

Why do the public transactions hurt the privacy of anon transactions? I don't get that part. An anon transactions can use only anon addresses as decoys no? So basically the anon coins and public coins are separated. Or am I missing something?

**Comment by particlmike33** [comment]
> No I don't disagree with that. But I think once the dex and marketplace are fully used we have enough anon transactions to be very private. It's up to the users to choose the amount of decoys/privacy. By only allowing anon transactions we can not use the governance system anymore for example.

@coybigpart, that is a future hypothetical situation with a really big "if".  And what are we to do in the present?  How are we to get more people interested in particl if it is not private currently?

And even if your hypothetical situation in the future ends up being true, and the % of public transactions is very small, why even allow them at that point if no one is using them?

And if we still do allow them in your hypothetical future situation, what if one day there are too many public transactions and ringct can be traced for that one day?  Is someone going to constantly monitor how many public transactions there are, and if too many alert everyone that ringct is not private for that day?

This all seems incredibly complicated and pinning our hopes on a lot of big "ifs" when instead we could just mandate ringct and be assured that we are actually private.  That is what the researchers in the paper suggest...

**Comment by particlmike33** [comment]
@coybigpart that's a good question, and I'm not a professional cryptographer.

All I know is that professional cryptographers have published peer reviewed papers that say it DOES affect ringct privacy:

"We first make the following observation. Consider the case, where,
a user Alice creates a transaction Tx-a with an input that she spends
without using any mix-in. Alice’s input key can clearly be identified
as spent. Now, consider a later transaction Tx-b created by another
user Bob who uses Alice’s input key as a mix-in for his input. If
the number of mix-ins used by Bob is one. Then, any adversary
looking at the blockchain can identify the real input key being
spent in Bob’s transaction and his input becomes traceable too.
More generally, if the number of mix-ins used by Bob is m > 1,
then the effective anonymity-set size now becomes m (instead of
the expected value of m + 1).
A closer look reveals that use of zero mix-in leads to a cascade
effect where the traceability of an input affects the traceability of
another input in a later transaction. Figure 6 schematically presents
this cascade effect triggered by Tx-a that uses no mix-in. Input in
Tx-a is clearly traceable. Tx-b uses one mix-in in the form of the
previously traced input and hence is also traceable. Tx-c now uses
two mix-ins as the real inputs of Tx-a and Tx-b respectively. Since,
these inputs have been previously identified as spent, Tx-c is also
traceable."

I think it would be prudent to listen to these professional security researchers that discovered this flaw back in 2017, that monero listened to their recommendation of implementing mandatory ringct, and we should also listen to the experts instead of just closing our eyes to the issue at hand.

**Comment by coybigpart** [comment]
@particlmike33 Is particl using public inputs as anon decoys?

**Comment by kangoala** [comment]
@particlmike33 
You're heading for the maximum privacy and ignoring that there are also other points to take care of. While BTC tries to maximize security and decentralization, it still lacks in terms of scalability. Same applies also for privacy coins, the anon set is not the only measure. You're certainly correct that anonymity is the most important measure, but if it would be the only one, I wouldn't stay with particl but rather with Monero. While Monero is PoW, Particl is PoS. While Monero can't process in an easy way Smart Contracts, Particl can do by their blinded coin states. The public coin state gives Particl a perfect match to the implemented and still growing BTC ecosystem. Everywhere where the infrastructure supports BTC, Particl could be set up as well. Finally it's up to each individual users whether to trust a given Particl anon set, or whether rather going with Monero or Zcash. In my opinion Particl is to Monero what Ethereum is to Bitcoin. When Ethereum is aiming for cheap transactions and high scalability it's becoming less decentralized. Ethereum is trying to complement Bitcoin and not replacing it. I think same is valid for Particl and Monero.

**Comment by particlmike33** [comment]
> @particlmike33 You're heading for the maximum privacy and ignoring that there are also other points to take care of. While BTC tries to maximize security and decentralization, it still lacks in terms of scalability. Same applies also for privacy coins, the anon set is not the only measure. You're certainly correct that anonymity is the most important measure, but if it would be the only one, I wouldn't stay with particl but rather with Monero. While Monero is PoW, Particl is PoS. While Monero can't process in an easy way Smart Contracts, Particl can do by their blinded coin states. The public coin state gives Particl a perfect match to the implemented and still growing BTC ecosystem. Everywhere where the infrastructure supports BTC, Particl could be set up as well. Finally it's up to each individual users whether to trust a given Particl anon set, or whether rather going with Monero or Zcash. In my opinion Particl is to Monero what Ethereum is to Bitcoin. When Ethereum is aiming for cheap transactions and high scalability it's becoming less decentralized. Ethereum is trying to complement Bitcoin and not replacing it. I think same is valid for Particl and Monero.

Interesting points @kangoala.  I agree with you that anonymity and privacy are the most important aspect of privacy, and without that our whole selling point kind of falls apart.

I don't think we need to give parts of our project up to increase our privacy, so why not do it?

Your worried about CEXs and compatibility when listing particl if we were to mandate ringct, but CEXs managed to list monero, and it's not like we are on many exchanges right now anyway.

Higher up, I stated that blind escrow transactions are important for programability, which is why the MP allows blinded escrow but enforces ringct on both sides of the escrow transaction.  And all "normal" transactions are forced to be ringct.

Why not adopt the rules of the MP for off MP transactions?  What would we lose exactly?  We'd gain better privacy and more respect among other privacy coins, and people would be more likely to use us and come to our community without opt-in privacy.

MP would still work and the DEX would still work.

What's to lose?

**Comment by particlmike33** [comment]
> @particlmike33 Is particl using public outputs as anon decoys?

@coybigpart Good question, I don't know, I'm not a dev or cryptographer.  I defer to the experts, and I know they found a flaw in opt-in ringct which monero chose to fix after the flaw was discovered, and we implemented monero's tech onto the bitcoin codebase from before they fixed their flaw.

Logic would entail that we have the same flaw then

**Comment by particlmike33** [comment]
@coybigpart even if the outputs are separated (can someone confirm?) the issue still remains that there needs to be enough ringct transactions for ringct to be private.

While it is definitely better if the outputs are separated, it doesn't completely resolve the situation.

Zcash has an anonymity set of all anonymous coins (unlike ring sigs which are smaller) because it uses zk-snarks.  And even zcash can be traced by chainalysis, because not enough of a % of all txs/day are anonymous txs.

If this issue exists for zk-snarks, it especially exists for ringct.

I'm not advocating for zk-snarks (don't like the trusted setup and the "moon math" lol), but just pointing out that separating the outputs isn't enough to solve the problem.

Every public transactions is taken away from a possible ringct transaction, which would help our privacy.

People have expressed concerns that taproot and LN aren't compatible with ringct.  Are they compatible with ct?

Perhaps people will be more amenable to mandating blind or ringct transactions (getting rid of public) instead of mandating just ringct, and blind only for escrow transactions--like the MP currently does

**Comment by coybigpart** [comment]
@particlmike33  I will answer the question for you. Public outputs can not be used as decoys. I suggest you do some more research about Particl or consult a cryptographer before doing a proposal you don't understand.
For me it's case closed now.

**Comment by kangoala** [comment]
@particlmike33 
I am also neither a developer nor a smart cryptographer, but understanding the basic principles of how the Particl blockchain creates its anon set is important when you are trying to defend your prospects so intensely.
I originally only used BTC so I feel most familiar with Particl's public ledger. During the hack I had a few frozen anon coins, but it didn't feel too convenient. When sending anon coins, you have to wait quite a long time for the order to be confirmed. Public transactions are much faster. On a public blockchain, you can also use a simple block explorer to check if the coins are still at the address they belong to. I like having the choice between public and anon coins depending on your usecase. If you want to anonymise your coins, you can always move them to anon and back. I know that's not in line with maximizing privacy, but since I don't plan to do any illegal things in the marketplace, I'm comfortable with a "not so perfect" anon set as well. I think particl can become successful even without ultra-max-privacy. As soon as the marketplace and BasicSwap take off, the anon set will also get better and we'll get closer to a Monero-like anonymity, which of course I'm striving for :-)
The Particl team is very talented and follows ideals that I share. In fact, I would always prefer to follow the opinion of technovert rather than hundreds of cryptographic papers if I don't know what ideals their authors follow.

**Comment by particlmike33** [comment]
@coybigpart that makes the situation less bad, but doesn't change the fact that public transactions take away transactions that could be ringct, and ringct needs a lot of transactions to function properly.  Every transaction that is public, would be ringct.

And there won't be many ringct transactions because it is opt-in, and by default it is public.

**Comment by particlmike33** [comment]
@kangoala I think without secure privacy, it hurts our MP and DEX.  We badly need more community members and generally wider knowledge of the particl project even existing.

And we've already seen monero people excited about our project's MP and DEX, only to be turned away when they find out we have opt-in privacy, because they are uninterested in a project that doesn't have real privacy.

I don't see the downside of at least mandating ct/anon transactions, instead of mandating anon only like this proposal was originally.

@coybigpart even with outputs separate, it doesn't change the fact that people can correlate amounts that are traceable in public transactions, and correlate that with the anon transactions.

With mandated ct/anon, this would be resolved

**Comment by kangoala** [comment]
@particlmike33 I totally agree that secure privacy is very, very important. I just wanted to point out that if I want to exchange Part with friends or pay with Part at the bakery, it would be much easier to realise this with public coins (as it is almost the same as BTC). I am already very confident that the other party would not be able to de-anonymise my personal Part accounts, provided the Part exchanged is from anon funds. 
Certainly, the Monero anon set is currently much better than Particl, and therefore it is not time (at the moment) to convince Monero followers of our privacy tech. Rather, we should try to convince average BTC users that the option to add more privacy features is better than just relying on pseudonymity.

**Comment by particlmike33** [comment]
@kangoala I see your point, but I'm concerned most people that would be interested in us care about privacy, and will be turned off by opt-in privacy.

Having only blind and anon transactions might be the best of both worlds.  We keep all our functionality and compatibility with LN and taproot, and we'd gain more privacy, and in addition lose a possible attack vector where amounts can be traced to specific people on the public side, and then correlated to the anon side.

With only blind (ct) that wouldn't be possible because amounts are hidden.

And realistically we are far off from in person companies accepting part, we need more people coming into our community first, and to have higher volume etc, which only having ct and anon could help because we'd be seen as more secure.

**Comment by particlmike33** [comment]
Closing this as @cryptoguard made some great points in discord.

Created a new proposal for easier reading and to reflect those new issues.

It can be found here:

https://github.com/particl/ccs-proposals/pull/14
